manual/source/classes/contributed_definitions/em-structure.rst

```diff
@@ -0,0 +1,134 @@
+
+    :ref:`NXpeak`:
+        A base class to describe peaks mathematically so that it can be used to detail how peaks in mass-to-charge-state ratio histograms (aka mass spectra) are defined and labelled as iontypes.
+
+    :ref:`NXpump`:
+        A base class to describe details about a pump in an instrument.
+
+    :ref:`NXscanbox_em`:
+        A base class to represent the component of an electron microscope which realizes a controlled deflection (and eventually shift, blanking, and/or descanning) of the electron beam to illuminate the specimen in a controlled manner. This can be used to document the scan pattern.
+
+    :ref:`NXspectrum_set`:
+        Base class and specializations comparable to NXimage_set but for storing spectra. Specialized base classes should use controlled vocabulary items as prefixes such as **eels** electron energy loss spectroscopy, **xray** X-ray spectroscopy (EDS/STEM, EDX, SEM/EDX, SEM/EDS), **auger** Auger spectroscopy, or **cathodolum** for cathodoluminescence spectra.
+
+    :ref:`NXstage_lab`:
+        As it was mentioned for atom probe microscopy, this is a base class to describe the stage/specimen holder which offers place for the documentation of the small-scale laboratory functionalities which modern stages of electron microscopes frequently offer.
+
+
+.. _EmPartnerClasses:
+
+Partner application definitions
+###############################
+
+A partner application definition is considered an application definition which stores data and metadata which are relevant for a given experiment but have usually only few connections to the detailed description of the workflow and experiment which motivates to granularize these pieces of information in an own application definition. In fact, one limitation of application definitions in NeXus is that they define a set of constraints on their graph of controlled concepts and terms. If we take for example diffraction experiments with an electron microscope it is usually the case that (diffraction) patterns are collected in the session at the microscope but all scientifically relevant conclusions are drawn later, i.e. through post-processing these data. These numerical and algorithmic steps define computational workflows were data from the application definitions such as NXem are used as input but many additional concepts and constraints may apply without any need for changing constraints on fields or groups of NXem. If we were to modify NXem for these cases, NXem would likely combinatorially diverge as every different combination of required constraints demands having an own but almost similar application definition. For this reason, the concept of partner application definition which have fields/links where specifically relevant sources of information are connected to e.g. NXem.
+More consolidation through the use of NXsubentry classes should be considered in the future.
```

Comment:
This pull request is adding a new "term of art" to NeXus, that of "Partner application definitions".  I'm not totally following from the description though.  @sanbrock can you restate what is going on with these?  Are there enough to justify a new NeXus term of art?

Reply:
It is good that you raise the question here whether it would need a new technical term. I am not convinced that this is really needed, my point was just that there are practical cases where to describe a workflow of measurement and analyses steps there are substantiated reasons for not having all pieces of information granularized into one appdef / base class but use a number of appdef i.e. there is some logical connection between different appdefs at least in how these may be used in certain fields.
I follow your suggestion and drop the term, instead I thought hard about how to connect different sets of granularized pieces of information in the field of EM.
The idea is essentially as follows and this is reflected in the refactored EM proposal:
NXem is the head appdef which specifies the arity and existence constraints which groups fields etc from a base class should be used, NXem_base is a proposal how one can use a base class to build a tree to relevant building blocks and base class instances to become together as a set useful and at least meaty enough to keep data, metadata and components of an EM and EM research. Initially we proposed base classes like image_secondary electron but there is nothing which makes an image secondary electron different from image backscattered electron, only the fact that what one sees in the image is different. Realizing this we now go for a concept/base class image and rather hook this base class into a deep base class e.g. NXem_eds which builds a tree to detail e.g. typical pieces of information worth to be documented for EDS energy-dispersive X-ray spectroscopy.
This NXem_eds base class is hooked-in in turn then into NXem_base.
The main idea behind NXem_base is thus to offer a template that still can be used when the only difference between a usage of an appdef is that several fields are required or optional because otherwise one would strictly speaking need as many appdefs as one has differly constraints trees how base classes are composed in EM research.

Reply:
Edited also in PR1271 now